### ISSUE TITLE
Endless migrations with .ValueGeneratedOnAddOrUpdate + .HasConversion column

### ISSUE THREAD

**Issue by avestnik** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#1757]
### Steps to reproduce
I need to map mysql timestamp field with default value and on update clause to long. So I added the following definitions for the Tester.Ticks field

``` c#
public class Tester
{
    public int Id { get; set; }
    public long Ticks { get; set; } // map to timestamp
}

public class TestContext : DbContext
{
    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    {
        optionsBuilder.UseMySql("server=locahost;database=db;user=user;password=1", new MySqlServerVersion("8.0.32-mysql"));
    }

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        base.OnModelCreating(modelBuilder);

        var e = modelBuilder.Entity<Tester>();

        e.Property(t => t.Id).ValueGeneratedOnAdd();
        e.HasKey(t => t.Id);

        e.Property(t => t.Ticks)
            .ValueGeneratedOnAddOrUpdate()
            .HasColumnType("TIMESTAMP")
            .HasPrecision(3)
            .HasConversion<DateTime>(s => new DateTime(s), date => date.Ticks)
            .HasDefaultValueSql("current_timestamp(3) ON UPDATE current_timestamp(3)")
            ;
    }

    public DbSet<Tester> Tests { get; set; }
}
```
solution with the above code is attached
[PomeloWithConversion.zip](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/files/10861146/PomeloWithConversion.zip)

### The issue
after adding initial migration all subsequent migrations include this:
``` c#
 public partial class Init2 : Migration
    {
        /// <inheritdoc />
        protected override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.AlterColumn<DateTime>(
                name: "Ticks",
                table: "Tests",
                type: "TIMESTAMP(3)",
                precision: 3,
                nullable: false,
                defaultValueSql: "current_timestamp(3) ON UPDATE current_timestamp(3)",
                oldClrType: typeof(DateTime),
                oldType: "TIMESTAMP(3)",
                oldPrecision: 3,
                oldDefaultValueSql: "current_timestamp(3) ON UPDATE current_timestamp(3)")
                .OldAnnotation("MySql:ValueGenerationStrategy", MySqlValueGenerationStrategy.ComputedColumn);
        }

        /// <inheritdoc />
        protected override void Down(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.AlterColumn<DateTime>(
                name: "Ticks",
                table: "Tests",
                type: "TIMESTAMP(3)",
                precision: 3,
                nullable: false,
                defaultValueSql: "current_timestamp(3) ON UPDATE current_timestamp(3)",
                oldClrType: typeof(DateTime),
                oldType: "TIMESTAMP(3)",
                oldPrecision: 3,
                oldDefaultValueSql: "current_timestamp(3) ON UPDATE current_timestamp(3)")
                .Annotation("MySql:ValueGenerationStrategy", MySqlValueGenerationStrategy.ComputedColumn);
        }
    }
```

which is kind of redundant and pollute the code. 

Please advise if it is pomelo or ef core related issue and if there is any workaround?

### Further technical details

MySQL version: 8.0
Operating system: 
Pomelo.EntityFrameworkCore.MySql version: 7.0.0
Microsoft.AspNetCore.App version: 7.0

Other details about my project setup:

**Comment by CliveBennett** [comment]
Yes, I have just discovered this issue also. No matter how many times I create a Migration and Deploy the Migration. Any migration I create now always has a AlterColumn Migration for the Timestamp column like this for every table.

 migrationBuilder.AlterColumn<DateTime>(
                name: "Timestamp",
                table: "Setting",
                type: "timestamp(6)",
                rowVersion: true,
                nullable: true,
                oldClrType: typeof(DateTime),
                oldType: "timestamp(6)",
                oldRowVersion: true,
                oldNullable: true)
                .Annotation("MySql:ValueGenerationStrategy", MySqlValueGenerationStrategy.ComputedColumn);
                
If I need to do an actual schema change. i'm just deleting all of these AlterColumn migrations, they seem to run ok.

I also note the ModelSnapshot doesnt get updated as part of the add-migration, IF there are no other schema changes.

**Comment by CliveBennett** [comment]
Thanks, good workaround, yes that does stop the Migrations, Ill add it for every table.

**Comment by lauxjpn** [comment]
We fixed this issue as part of #1857/#1869.

---

> Thanks, good workaround, yes that does stop the Migrations,
> 
> I implemented like this so it does every table (as I have timestamp in a base class)
> 
> ```
> foreach (var entityType in modelBuilder.Model.GetEntityTypes())
> {
>     var timestampProperty = entityType.FindProperty("Timestamp");
> 
>     if (timestampProperty != null)
>     {
>         timestampProperty.AddAnnotation("MySql:ValueGenerationStrategy", MySqlValueGenerationStrategy.ComputedColumn);
>     }
> }
> ```

If you used this workaround, you likely want to remove it when upgrading to `8.0.2`, or it might have some unintended side effects. See https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/1865#issuecomment-1986030983 for more information.
